doc/rest-api.md

```diff
@@ -1818,3 +1818,287 @@ Input (none at present):
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+ Input:
+
+    {
+        "config": {
+            "size": "15032385536",
+            "source": "pool1",
+            "volume.block.filesystem": "xfs",
+            "volume.block.mount_options": "discard",
+            "volume.lvm.thinpool_name": "LXDPool",
+            "volume.size": "10737418240"
+        }
+    }
+
+### PATCH
+
+ * Description: update the storage pool configuration
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {
+            "volume.block.filesystem": "xfs",
+        }
+    }
+
+### DELETE
+
+ * Description: delete a storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input (none at present):
+
+    {
+    }
+
+## /1.0/storage-pools/<pool_name>/volumes
+
+### GET
+
+ * Description: list of storage volumes
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of storage volumes that currently exist on a given storage pool
+
+    [
+        {
+            "volume_type": "container",
+            "used_by": [],
+            "volume_name": "alp1/snap1",
+            "config": {}
+        },
+        {
+            "volume_type": "container",
+            "used_by": [],
+            "volume_name": "alp1/snap2",
+            "config": {}
+        },
+    ]
+
+## /1.0/storage-pools/<pool_name>/volumes/<volume_type>
+
+### POST
+
+ * Description: create a new storage volume on a given storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {},
+        "pool_name": "pool1",
+        "volume_name": "vol1",
+        "volume_type": "custom"
+    }
+
+
+## /1.0/storage-pools/<pool_name>/volumes/<volume_type>/<volume_name>
+
+### GET
+
+ * Description: information about a storage volume of a given type on a storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: dict representing a storage volume
+
+    {
+        "type": "sync",
+        "status": "Success",
+        "status_code": 200,
+        "operation": "",
+        "error_code": 0,
+        "error": "",
+        "metadata": {
+            "volume_type": "custom",
+            "used_by": [],
+            "volume_name": "vol1",
+            "config": {
+                "block.filesystem": "ext4",
+                "block.mount_options": "discard",
+                "lvm.thinpool_name": "LXDPool",
+                "size": "10737418240"
+            }
+        }
+    }
+
+
+### POST
+
+ * Description: rename a storage volume of a given type on a storage pool
+ - Currently not implemented.
```

Comment:
If not implemented, then remove it and mention in the github issue about pool renames (make it cover both pool and volume renames).